pkg/deployer/helm/ensure.go

```diff
@@ -162,10 +162,25 @@ func (h *Helm) createOrUpdateExport(ctx context.Context, values map[string]inter
 }
 
 func (h *Helm) DeleteFiles(ctx context.Context) error {
+	helm.log.Info("Deleting files")
 	h.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseDeleting
 	status := &helmv1alpha1.ProviderStatus{}
-	if _, _, err := serializer.NewCodecFactory(Helmscheme).UniversalDecoder().Decode(h.DeployItem.Status.ProviderStatus.Raw, nil, status); err != nil {
-		return err
+
+	factory := serializer.NewCodecFactory(Helmscheme)
+	if &factory == nil {
+		return fmt.Errorf("Cannot create factory for helm scheme %s", Helmscheme)
+	}
+
+	decoder := factory.UniversalDecoder()
+	if &decoder == nil {
+		return errors.New("Cannot get decoder from factory")
+	}
+
+	if h.DeployItem.Status.ProviderStatus != nil {
+		h.log.Info(fmt.Sprintf("Provider status is: %s", h.DeployItem.Status.ProviderStatus))
+		if _, _, err := decoder.Decode(h.DeployItem.Status.ProviderStatus.Raw, nil, status); err != nil {
+			return err
+		}
```

Comment:
the factory and the decoder can never be nil therefore the check is not needed.
```suggestion
	if h.DeployItem.Status.ProviderStatus != nil {
	        serializer.NewCodecFactory(Helmscheme).UniversalDecoder()
		h.log.Info(fmt.Sprintf("Provider status is: %s", h.DeployItem.Status.ProviderStatus))
		if _, _, err := decoder.Decode(h.DeployItem.Status.ProviderStatus.Raw, nil, status); err != nil {
			return err
		}
```

Reply:
Applied.